### PR TITLE
docs: rego rule explanation and links

### LINKED ISSUE #516: Document where to find production ready rego rules
```[tasklist]
### Tasks
- [x] Link to hdfs-utils repo
- [x] Link to integration tests folder
- [x] Provide a short example and explanation of the example
```

### PR DESCRIPTION
**Pull request by adwk67**
# Description

fixes https://github.com/stackabletech/hdfs-operator/issues/516.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Link to hdfs-utils repo
- [x] Link to integration tests folder
- [x] Provide a short example and explanation of the example
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: docs/modules/hdfs/pages/usage-guide/security.adoc [hunk 1]
```diff
@@ -69,7 +69,9 @@ include::example$usage-guide/hdfs-regorules.yaml[]
 ----
 
 This rego rule is intended for demonstration purposes and allows every operation.
-For a production setup you probably want to take a look at our integration tests for a more secure set of rego rules.
+For a production setup you will probably need to have something much more granular.
+We provide a more representative rego rule in our integration tests and in the afore-mentioned hdfs-utils repository.
```
Comment (sbernauer) [inline docs/modules/hdfs/pages/usage-guide/security.adoc]: Had to google this word, IDK if this is correct
```suggestion
We provide a more representative rego rule in our integration tests and in the aforementioned hdfs-utils repository.
```

**Review by sbernauer — DISMISSED** [review]
This exceeds what I expected to document, so many thanks :)

I'm not 100% sure if we want to link to the rego rules in hdfs-utils *and* integration tests. I think it's a bit easier to only have a single source of truth - which needs to be the integration test to align with trino-operator

**Comment by adwk67** [comment]
> This exceeds what I expected to document, so many thanks :)
> 
> I'm not 100% sure if we want to link to the rego rules in hdfs-utils _and_ integration tests. I think it's a bit easier to only have a single source of truth - which needs to be the integration test to align with trino-operator

Yes, good point. I'd quite like to keep the reference to `hdfs_test.rego` in there, which refers to the rego rule in hdfs-utils. I can maybe just use it in-line and remove the link to hdfs-utils. What do you think?

**Comment by sbernauer** [comment]
Ahh I see the point of including the test :+1: Yep, that suggestion sounds good to me :+1:

File: docs/modules/hdfs/pages/usage-guide/security.adoc [hunk 2]
```diff
@@ -109,6 +111,140 @@ The implication is thus that you cannot add users to the `superuser` group, whic
 We have decided that this is an acceptable approach as normal operations will not be affected.
 In case you really need users to be part of the `superusers` group, you can use a configOverride on `hadoop.user.group.static.mapping.overrides` for that.
 
+[#_fine_granular_rego_rule_]
+=== Fine-granular rego rule
```
Comment (fhennig) [inline docs/modules/hdfs/pages/usage-guide/security.adoc]:
```suggestion
[#fine-granular-rego-rules]
=== Fine-granular rego rules
```

File: docs/modules/hdfs/pages/usage-guide/security.adoc [hunk 3]
```diff
@@ -69,7 +69,9 @@ include::example$usage-guide/hdfs-regorules.yaml[]
 ----
 
 This rego rule is intended for demonstration purposes and allows every operation.
-For a production setup you probably want to take a look at our integration tests for a more secure set of rego rules.
+For a production setup you will probably need to have something much more granular.
+We provide a more representative rego rule in our integration tests and in the aforementioned hdfs-utils repository.
+Details can be found below in the <<_fine_granular_rego_rule_, fine-granular rego rule>> section.
```
Comment (fhennig) [inline docs/modules/hdfs/pages/usage-guide/security.adoc]:
```suggestion
Details can be found below in the <<fine-granular-rego-rules, fine-granular rego rules>> section.
```

File: docs/modules/hdfs/pages/usage-guide/security.adoc [hunk 4]
```diff
@@ -109,6 +111,140 @@ The implication is thus that you cannot add users to the `superuser` group, whic
 We have decided that this is an acceptable approach as normal operations will not be affected.
 In case you really need users to be part of the `superusers` group, you can use a configOverride on `hadoop.user.group.static.mapping.overrides` for that.
 
+[#_fine_granular_rego_rule_]
+=== Fine-granular rego rule
+
+The hdfs-utils repository contains a more production-ready rego-rule https://github.com/stackabletech/hdfs-utils/blob/main/rego/hdfs.rego[here].
+With a few minor differences (e.g. pod names) it is the same rego rule that is used in this https://github.com/stackabletech/hdfs-operator/blob/main/tests/templates/kuttl/kerberos/12-rego-rules.txt.j2[integration test].
```
Comment (fhennig) [inline docs/modules/hdfs/pages/usage-guide/security.adoc]:
```suggestion
With a few minor differences (e.g. Pod names) it is the same rego rule that is used in this https://github.com/stackabletech/hdfs-operator/blob/main/tests/templates/kuttl/kerberos/12-rego-rules.txt.j2[integration test].
```

**Review by fhennig — DISMISSED** [review]

**Comment by sbernauer** [comment]
> Yes, good point. I'd quite like to keep the reference to hdfs_test.rego in there, which refers to the rego rule in hdfs-utils. I can maybe just use it in-line and remove the link to hdfs-utils. What do you think?

Thinking about this, in trino-operator we have [the rego rules *and* the tests](https://github.com/stackabletech/trino-operator/tree/main/tests/templates/kuttl/opa-authorization/trino_rules). We could copy/paste the approach to [hdfs-operator](https://github.com/stackabletech/hdfs-operator/tree/main/tests/templates/kuttl/kerberos), so that we can link to rego rules and tests with a single link.

I can volunteer to copy the setup from trino to hdfs, pls ping me if I should do it in this branch

**Comment by adwk67** [comment]
> I can volunteer to copy the setup from trino to hdfs, pls ping me if I should do it in this branch

That would be great @sbernauer : thank you!

**Review by sbernauer — APPROVED** [review]
> I can volunteer to copy the setup from trino to hdfs, pls ping me if I should do it in this branch

After spending quite some time on it I think I would drop this. In comparison to trino we need to a.) run jinja template and b.) substitute the namespace when applying.
When doing all of this we loose the benefit of being to `opa test` it...

So I would say it's totally fine to link to the integration test for the source of truth of rego rules (they are nightly tested) and link to hdfs-utils for the rego rule tests in case someone is interested

**Pull request merged**